lib/srv/db/cloud/resource_checker_url_aws.go

```diff
@@ -272,3 +272,30 @@ func (c *urlChecker) checkOpenSearchEndpoint(ctx context.Context, database types
 		return trace.BadParameter(`cannot validate OpenSearch custom domain %v. Please provide Database Service "es:DescribeDomains" permission to validate the URL.`, database.GetURI())
 	}
 }
+
+func (c *urlChecker) checkDocumentDB(ctx context.Context, database types.Database) error {
+	meta := database.GetAWS()
+	rdsClient, err := c.clients.GetAWSRDSClient(ctx, meta.Region,
+		cloud.WithAssumeRoleFromAWSMeta(meta),
+		cloud.WithAmbientCredentials(),
+	)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	cluster, err := describeRDSCluster(ctx, rdsClient, meta.DocumentDB.ClusterID)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	databases, err := common.NewDatabasesFromDocumentDBCluster(cluster)
+	if err != nil {
+		c.log.Warnf("Could not convert DocumentDB cluster %q to database resources: %v.",
+			aws.StringValue(cluster.DBClusterIdentifier), err)
+
+		// services.NewDatabasesFromDocumentDBCluster maybe partially successful.
```

Comment:
Can you expand on why we want to return nil here?
```suggestion
		// common.NewDatabasesFromDocumentDBCluster may be partially successful.
```

Reply:
it shouldn't. will fix. thanks!